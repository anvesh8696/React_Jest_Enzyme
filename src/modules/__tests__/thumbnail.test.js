import React from 'react';
import { shallow, mount } from 'enzyme';
import { spy } from 'sinon';
import Portal from 'react-portal';
import fakePhotos from '../__data__/photos.fake';
import toJson from 'enzyme-to-json';
import Thumbnail from '../thumbnail';


describe('<Thumbnail />', () => {
	let wrapper, instance;
	let onClickSpy, openPortalSpy;


	beforeEach(() => {
		onClickSpy = spy(Thumbnail.prototype, 'onClick');
		openPortalSpy = spy(Portal.prototype, 'openPortal');
		wrapper = mount(<Thumbnail photo={fakePhotos[0]} />);
		instance = wrapper.instance();
	});

	afterEach(() => {
		Thumbnail.prototype.onClick.restore();
		Portal.prototype.openPortal.restore();
	});

	it('responds to a click', () => {
		wrapper.find('img').simulate('click');
		expect(onClickSpy.calledOnce).toBe(true);
	});

	it('opens a portal on click', () => {
		wrapper.find('img').simulate('click');
		expect(openPortalSpy.calledOnce).toBe(true);
	});

	it('matches snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

});