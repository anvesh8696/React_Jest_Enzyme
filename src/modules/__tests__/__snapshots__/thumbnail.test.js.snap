exports[`<Thumbnail /> matches snapshot 1`] = `
<Thumbnail
  photo={
    Object {
      "id": 0,
      "thumbnailUrl": "thumbnail0",
      "url": "fakePhoto0",
    }
  }>
  <div
    className="thumbnail">
    <img
      onClick={[Function]}
      src="thumbnail0" />
    <Portal
      beforeClose={[Function]}
      closeOnEsc={true}
      closeOnOutsideClick={true}
      onClose={[Function]}
      onOpen={[Function]}
      onUpdate={[Function]} />
  </div>
</Thumbnail>
`;
